### DISCUSSION TITLE
add "healthy" sdnotify policy

### DISCUSSION THREAD

**Discussion by maciejpajak** [containers/podman#19380] [General]
Hi, when is the next release planed containing /add "healthy" sdnotify policy/ feature?

**Comment by vrothberg** [comment]
Thanks for reaching out, @maciejpajak !

Since Podman went out just last week, I don't expect a 4.7 release before late September.

**Reply by maciejpajak** [reply]
Thank you @vrothberg

**Comment by maciejpajak** [comment]
@vrothberg,  I have one more question. Is there possibibility for current release to set a timeout for sending "systemd-notify --ready" from working container when the option "--sdnotify=container" is set of "podman run" command before "podman auto-update" starts rollback ?

**Reply by vrothberg** [reply]
New features will not be backported to releases but only bugs.  Can you elaborate on the problem you want to solve?  Maybe we can find a workaround.

**Reply by maciejpajak** [reply]
I am trying to implement "podman auto-update". I generated the systemd unit and set --sdnotify=container for the podman run command in the unit's ExecStart file. The auto-update succeeds when the "systemd-notify --ready" command is run on the container at the beginning of the main container process. But to decide if my container is working fine, I have to run "systemd-notify --ready" about 30 seconds later after the container starts. This causes auto-update to start a "rollback" and not wait for "systemd-notify" from the container.

**Reply by maciejpajak** [reply]
`"healthy" sdnotify policy` combines the functionality of health-check and auto-update and this would be a great solution to my problem - I suppose. But this time I have to find workaround.

**Reply by maciejpajak** [reply]
@vrothberg I think I found a solution. A TimeoutSec=30 in [Service] section of the systemd unit file does a job. Thank you!
